stdlib/asm/crypto/stark/deep_queries.masm

```diff
@@ -873,597 +490,184 @@ export.compute_deep_composition_polynomial_queries
-
-        ext2add
-        #=> [accH_1, accH_0, p2, p1, Acc, Z, y, ptr_query ]
-        #=> [accH_1, accH_0, p2, p1, acc_z_1, acc_z_0, acc_gz_1, acc_gz_0,      Z ]
-
-        # Divide by (x - z), (x - gz)
-        swapw.2
         swapw
-        #=> [acc_z_1, acc_z_0, acc_gz_1, acc_gz_0, (x - z)_1, (x - z)_0, (x - gz)_1, (x - gz)_0, accH_1, accH_0, p2, p1, y, ptr_query ]
-
-        movup.5
-        movup.5
-        ext2div
-        #=> [query_z_1, query_z_0, acc_gz_1, acc_gz_0, (x - gz)_1, (x - gz)_0, accH_1, accH_0, p2, p1, y, ptr_query ]
-        # where (query_z_0, query_z_1) = (acc_z_0, acc_z_1)/(x - z)
-
-        movdn.5
-        movdn.5
-        #=> [acc_gz_1, acc_gz_0, (x - gz)_1, (x - gz)_0, query_z_1, query_z_0, accH_1, accH_0, p2, p1, y, ptr_query ]
-
-        movup.3
-        movup.3
-        ext2div
-        #=> [query_gz_1, query_gz_0, query_z_1, query_z_0, accH_1, accH_0, p2, p1, y, ptr_query ]
-
-        # Sum the two accumulators to get the part of the query comming from the traces
-        ext2add
-        #=> [query_trace_1, query_trace_0, accH_1, accH_0, p2, p1, y, ptr_query ]
-
-        # Divide the constraint composition polynomial accumulator by (x - z^8)
-        ## Get x
+        #=> [Y, Acc, P, Z, ptr_query_end, ptr_query]
+        #=> where Acc =: [Acc3, Acc2, Acc1, Acc0]
+        
+        ## This, together with the pre-existing Y, will be used to mstream the elements T_i(x)
         push.0.0.0.0
-        mem_loadw.TMP2
-        movdn.3
-        push.0
-        mem_loadw.Z_PTR
-        #=> [z^8_1, z^8_0, z1, z0, x, query_trace_1, query_trace_0, accH_1, accH_0, p2, p1, y, ptr_query ]
-
-        ## Compute the denominator
-        neg
-        swap
-        movup.4
-        swap
-        sub
-        swap
-        #=> [-z^8_1, x - z^8_0, z1, z0, query_trace_1, query_trace_0, accH_1, accH_0, p2, p1, y, ptr_query ]
+        #=> [Y, Y, Acc, P, Z, ptr_query_end, ptr_query]
 
-        ## Divide by the computed denominator
-        movdn.5
-        movdn.5
-        swapw
-        #=> [-z^8_1, x - z^8_0, accH_1, accH_0, z1, z0, query_trace_1, query_trace_0, p2, p1, y, ptr_query  ]
+        ## Compute the random linear combination for the first 72 main trace columns
+        repeat.9
+            mem_stream
+            repeat.8
+                exec.combine_main
+            end
+        end
+        
+        ## and the 73rd main column
+        mem_stream
+        exec.combine_main
+
+        ## Compute the random linear combination of the first 8 auxiliary trace columns
+        repeat.2
+            mem_stream
+            repeat.4
+                exec.combine_aux
+            end
+        end
+        
+        ## and the 9th aux column
+        mem_stream
+        exec.combine_aux
+
+        ## Compute the random linear combination of the constraint composition trace part. 
+        repeat.2
+            mem_stream
+            #=> [H, J, Acc, P]
+            swapw.3
+            swapw
+            #=> [J, P, Acc, H]
```

Comment:
This and below code computes the equivalent of [this](https://github.com/facebook/winterfell/blob/main/verifier/src/composer.rs#L151) in Winterfell, right? If so, I'm wondering if we can re-use `combine_aux` procedure and just set irrelevant parts to `ZERO`. This will require some extra stack manipulation - so, not sure if it would be a net benefit (but seems like it could be).

Also, maybe we should create helper procedures for this and above code as well? For example, `compose_trace_columns` and `compose_constraint_evaluations`.

Reply:
Yes, I thought about it but the OOD part that is in the helper registers is the bottleneck because we would have to not use `adv_pipe` when loading it into memory so that we can lay it out as `[H_i(z)0, H_i(z)1, 0, 0]`.